### DISCUSSION TITLE
How to properly disable AVAHISCAN?

### DISCUSSION THREAD

**Discussion by RompitaKablo** [netalertx/NetAlertX#847] [Q&A]
I am posting here rather than creating a bug report as at this time I am not sure if the fault is with NetAlertX or my understanding of how it works.

I disabled the AVAHISCAN plugin and removed it from "Loaded plugins". However if I later save the settings for any reason the AVAHISCAN entry re-appears under "Other Scanners". 

![avahi_1](https://github.com/user-attachments/assets/15f67dab-4d0b-4786-bfb9-dbfc7c4fc665)

Also its settings section reappears, set as "before_name_updates" rather than "disabled". However the plugin is still not present in "loaded plugins". 

![avahi_2](https://github.com/user-attachments/assets/397803a0-77d1-49df-a3bb-fc3af3f3e298)

Am I doing this wrong?

**Comment by jokob-sk** [comment]
Hi @RompitaKablo ,

Setting a plugin to `disabled` and then removing it from the loaded plugins should remove it (unless it's a core plugin).

This is probably a caching issue. Did you clear the browser cache and clicked the 🔄 button in the header? This usually refresheds the enabled scanners section. 

Thanks, 
j

**Comment by RompitaKablo** [comment]
> Did you clear the browser cache and clicked the 🔄 button in the header? 

I did but as soon as I press "Save" in the settings, even if nothing has been changed, AVAHISCAN returns. To be clear, the tile reappears under "Other Scanners" and the settings panel reappears under "Other Scanners" (see original post) but the plugin does **not** reappear as a "Loaded Plugin".

AVAHISCAN is even shown in the progress box at the top right when the settings have been saved.

![avahiscan](https://github.com/user-attachments/assets/e7b3d39e-1731-42d0-b9a5-6d8a8c86f38a)

**Reply by jokob-sk** [reply]
I'm able to disable the scanner just fine:

![ezgif-3-6a3d0c43be](https://github.com/user-attachments/assets/38012acf-575e-47fc-8278-f9f66590ad1c)

You can ignore the message in the top progress box - it will be shown even if the scanner is disabled - I will look into that later. It requires some deeper rework of how name discovery plugins are handled.

**Comment by RompitaKablo** [comment]
Thank you for your reply. I can disable AVAHISCAN as well but after it is disabled if I ever save the settings again, for whatever reason, even if nothing has changed, it comes back again (except in the "Loaded Plugins" section).

There must be something wrong with my setup. I will at some point wipe the lot and start again.

**Reply by jokob-sk** [reply]
Hummm, just make sure you save your settings after disabling the scan, so:

1. Set AVAHISCAN to disabled
2. Save settings
3. Remove AVAHISCAN from LOADED_PLUGINS
4. Save settings

**Reply by jhillix** [reply]
I'm experiencing the same issue, so you are not alone, and I do not believe it's a setup issue or cache-related. Either this is expected behavior because of the previous comment about Core plugins - - OR -  - this is a bug. Either way, it's annoying.

**Reply by jokob-sk** [reply]
Hey, Just tried the above approach and it's still working to disable it this way:


1.     Set AVAHISCAN to disabled
2.     Remove AVAHISCAN from LOADED_PLUGINS
3.     Save settings

Could you please share your approach?

**Reply by jhillix** [reply]
Even after performing the steps above, `AVAHISCAN` is coming back. I think it's because I did Step 2 before Step 1 previously, and now it's in some funky state.

**Reply by jokob-sk** [reply]
Can you try the `netalertx-dev` image in about 15 min? I think there was a bug

**Reply by jhillix** [reply]
> Can you try the `netalertx-dev` image in about 15 min? I think there was a bug

What do you recommend I do to try `netalertx-dev`?

**Reply by jokob-sk** [reply]
The same approach highlighted  above should work in the dev image
